runtime/common/src/migrations/transfer_allowlist_currency.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_primitives::{AccountId, BlockNumber};
+use cfg_types::{locations::Location, tokens::CurrencyId};
+use frame_support::{
+	pallet_prelude::{NMapKey, OptionQuery},
+	storage::types::{StorageDoubleMap, StorageNMap},
+	traits::{Get, OnRuntimeUpgrade, StorageInstance},
+	weights::Weight,
+	Blake2_128Concat, Twox64Concat,
+};
+use pallet_transfer_allowlist::{AllowanceDetails, AllowanceMetadata};
+#[cfg(feature = "try-runtime")]
+use parity_scale_codec::{Decode, Encode};
+#[cfg(feature = "try-runtime")]
+use sp_runtime::DispatchError;
+
+use crate::transfer_filter::FilterCurrency;
+
+const LOG_PREFIX: &str = "TransferAllowlist-CurrencyMigration: ";
+
+struct DelayPrefix;
+impl StorageInstance for DelayPrefix {
+	const STORAGE_PREFIX: &'static str = "AccountCurrencyTransferCountDelay";
+
+	fn pallet_prefix() -> &'static str {
+		"TransferAllowList"
+	}
+}
```

Comment:
Out of curiosity, did you try `#[pallet::storage_prefix = "foo"]` this?

Reply:
No, you wrote me too late. ^^ 

I think I will keep for now, as I am not sure, how to add the pallet-prefix with this macro. and lazy to change the code. We will remove it anyways afterwards.